core/src/main/java/org/apache/iceberg/actions/BaseRewriteDataFilesAction.java

```diff
@@ -238,12 +239,13 @@ public RewriteDataFilesActionResult execute() {
     }
 
     List<DataFile> addedDataFiles = rewriteDataForTasks(combinedScanTasks);
-    List<DataFile> currentDataFiles = combinedScanTasks.stream()
+    Set<DataFile> currentDataFiles = combinedScanTasks.stream()
         .flatMap(tasks -> tasks.files().stream().map(FileScanTask::file))
-        .collect(Collectors.toList());
-    replaceDataFiles(currentDataFiles, addedDataFiles);
+        .collect(Collectors.toSet());
+    List<DataFile> deletedDataFiles = Lists.newArrayList(currentDataFiles.iterator());
```

Comment:
Do we need .iterator? I thought Lists.newArrayList took iterable as an arg, so set should be fine all by itself, although I'm not sure why we are doing the "set" operation